src/index.js

```diff
@@ -20,15 +20,100 @@ function createElement(tagName, props, ...children) {
   return element;
 }
 
-function render() {
+const initialState = {
+  previousValue: '0',
+  currentValue: '0',
+  currentOperator: '',
+  status: 'operand',
+};
+
+const sum = (a, b) => Number(a) + Number(b);
+const subtract = (a, b) => Number(a) - Number(b);
+const multiply = (a, b) => Number(a) * Number(b);
+const divide = (a, b) => Number(a) / Number(b);
+
+const calculate = (operator, a, b) => ({
+  '+': sum(a, b),
+  '-': subtract(a, b),
+  '*': multiply(a, b),
+  '/': divide(a, b),
+})[operator];
+
+function render(state) {
+  const {
+    previousValue, currentValue, currentOperator, status,
+  } = state;
+
+  function handleOperand(value) {
+    if (status === 'operator') {
+      return render({
+        ...state,
+        currentValue: value,
+        status: 'operand',
+      });
+    }
+
+    if (currentValue[0] === '0') {
+      return render({
+        ...state,
+        currentValue: value,
+        status: 'operand',
+      });
+    }
+
+    return render({
+      ...state,
+      currentValue: currentValue + value,
+      status: 'operand',
+    });
```

Comment:
```suggestion
    return render({
      ...state,
      currentValue: currentValue === '0' ? value : (currentValue + value),
      status: 'operand',
    });
```

중복되는 것을 제거하고, `'0'`이라는 기본값일 경우에는 value라고 줘볼려고 했는데 뭔가 더 복잡해진 것 같기도 하네요

Reply:
비슷한 로직을 반복한것보다 훨씬 짧아진것 같아서 좋은 것 같습니다 :)